components/klaudsolcms/modals/modal_body/CollectionTypeBody.js

```diff
@@ -1,65 +1,85 @@
-import { useState, useContext } from 'react';
-import {Formik, Form, Field} from 'formik';
-import { loadEntityTypes } from '@/components/reducers/actions';
-import RootContext from '@/components/contexts/RootContext';
+import { useState, useContext, useEffect } from "react";
+import { Formik, Form, Field, useFormikContext, useField } from "formik";
+import { loadEntityTypes } from "@/components/reducers/actions";
+import RootContext from "@/components/contexts/RootContext";
+import DependentField from "@/components/fields/DependentField";
 
-export default function CollectionTypeBody({formRef}) {
-    const { state: rootState, dispatch: rootDispatch } = useContext(RootContext);
+export default function CollectionTypeBody({ formRef }) {
+  const { state: rootState, dispatch: rootDispatch } = useContext(RootContext);
 
-    const formikParams = {
-      initialValues: {},
-      innerRef: formRef,
-      onSubmit: (values) => {
-        console.error(JSON.stringify(values));
-
-        (async () => {
-          try {
+  const formikParams = {
+    initialValues: {},
+    innerRef: formRef,
+    onSubmit: (values) => {
+      (async () => {
+        try {
           //refactor to reducers/actions
-           await fetch(`/api/entity_types`, {
-              method: 'POST',
-              body: JSON.stringify(values),
-              headers: {
-                'Content-Type': 'application/json'
-              }
-            });
-          } catch (error) {
+          await fetch(`/api/entity_types`, {
+            method: "POST",
+            body: JSON.stringify(values),
+            headers: {
+              "Content-Type": "application/json",
+            },
+          });
+        } catch (error) {
+        } finally {
+          await loadEntityTypes({ rootState, rootDispatch });
+        }
+      })();
+    },
+  };
+
+  const dependentFieldFormat = (value) => {
+    const valueSplit = value?.split(" ");
+    // Prevents extra spaces from becoming '-'
+    const filteredSplit = valueSplit?.filter((i) => i !== "");
+    const joinedSplit = filteredSplit?.join("-");
+
+    // If the user types in space as the first letter, a '-'
+    // will appear, the trim() method below will prevent it
+    if (value?.trim() === "") return "";
+    if (!value?.endsWith(" ")) return joinedSplit;
+
+    const newVal = `${joinedSplit}-`;
```

Comment:
Is there a reason why we have a terminal `-` in the slug? This to me looks unsightly.
```suggestion
    return joinedSplit;
```

Reply:
The purpose of that line is to add a `-` immediately every time a user inputs a space. Without this line, every time a user presses space, the `-` will not appear immediately. It will only appear once the user types in anything that is not a space.  
  
In the final output, the `-` would not exist, unless if the user specifically typed a space at the end of the name.

Reply:
Ok :)